### PR TITLE
Update WR (render notifier API changes).

### LINKED ISSUE #13480: intermittent crash ("Connection reset by peer" from create_api) in /html/semantics/embedded-content/the-canvas-element/2d.getcontext.shared.html, /html/semantics/scripting-1/the-template-element/template-element/node-document-changes.html, /dom/traversal/TreeWalker-walking-outside-a-tree.html, /XMLHttpRequest/responseText-status.html, /_mozilla/css/line_break_nowrap.html
```
 ▶ CRASH [expected FAIL] /css21_dev/html4/vertical-align-007.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  │ called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Os { code: 104, message: "Connection reset by peer" } }) (thread LayoutThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }, at ../src/libcore/result.rs:799)
  │ stack backtrace:
  │    0:     0x7fa759924c8d - backtrace::backtrace::trace::h0e60ef08c7c34e9f
  │    1:     0x7fa759925312 - backtrace::capture::Backtrace::new::h8bf319c36d8f5d1b
  │    2:     0x7fa75846e9cd - servo::main::_{{closure}}::h911afab4cc009f33
  │    3:     0x7fa759f03745 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
  │    4:     0x7fa759f03612 - std::panicking::begin_panic::hfbeda5aad583dc32
  │    5:     0x7fa759f03550 - std::panicking::begin_panic_fmt::h4fe9fb9d5109c4bf
  │    6:     0x7fa759f034d1 - rust_begin_unwind
  │    7:     0x7fa759f3ff5c - core::panicking::panic_fmt::h4395919ece15c671
  │    8:     0x7fa759a8964f - core::result::unwrap_failed::h3191a47cbfbe2a26
  │    9:     0x7fa759a8ad20 - webrender_traits::api::_<impl webrender_traits..RenderApiSender>::create_api::h74e2147a94b73b3e
  │   10:     0x7fa7587e10f7 - _<layout_thread..LayoutThread as layout_traits..LayoutThreadFactory>::create::_{{closure}}::hcbdf224d94fa1d3a
  │   11:     0x7fa75875c796 - std::panicking::try::do_call::h032f92c4e6f5b87e
  │   12:     0x7fa759f0c0c6 - __rust_maybe_catch_panic
  │   13:     0x7fa758783e44 - _<F as alloc..boxed..FnBox<A>>::call_box::h018a2ddf52f3d72e
  │   14:     0x7fa759f01d80 - std::sys::thread::Thread::new::thread_start::h022e3887023c6290
  │   15:     0x7fa756378183 - start_thread
  │   16:     0x7fa755e8f37c - clone
  │   17:                0x0 - <unknown>
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Os { code: 104, message: "Connection reset by peer" } })
  └ Pipeline failed in hard-fail mode.  Crashing!
```

### PR DESCRIPTION
**Pull request by glennw**
These changes fix #13480.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18986)

### CODE REVIEW

**Comment by glennw** [comment]
r? @jdm 

This includes your commit in https://github.com/servo/servo/pull/18976, and a follow up commit that (a) updates Cargo.lock (b) updates your commit with the final WR changes.

Your original commit looks good to me.

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -313,25 +311,29 @@ fn initialize_png(gl: &gl::Gl, width: usize, height: usize) -> RenderTargetInfo
...
         RenderNotifier {
             compositor_proxy: compositor_proxy,
         }
     }
 }
 
 impl webrender_api::RenderNotifier for RenderNotifier {
-    fn new_frame_ready(&mut self) {
+    fn clone(&self) -> Box<webrender_api::RenderNotifier> {
+        Box::new(RenderNotifier::new(self.compositor_proxy.clone()))
```
Comment (emilio) [inline components/compositing/compositor.rs]: Can this be just `Box::new(self.clone())`

**Review by emilio — APPROVED** [review]

File: components/compositing/compositor.rs (hunk 1 above)
Comment (jdm) [reply components/compositing/compositor.rs]: I feel like it becomes awkward calling self.clone() inside of a method named clone. I would rather leave it.

**Comment by jdm** [comment]
It looks like something changed related to borders since the last WR update:
```
  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-color-applies-to-003.htm
  └   → /css21_dev/html4/border-left-color-applies-to-003.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-applies-to-002.htm
  └   → /css21_dev/html4/border-left-applies-to-002.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-width-applies-to-003.htm
  └   → /css21_dev/html4/border-left-width-applies-to-003.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-color-applies-to-001.htm
  └   → /css21_dev/html4/border-left-color-applies-to-001.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-color-applies-to-002.htm
  └   → /css21_dev/html4/border-left-color-applies-to-002.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-width-applies-to-002.htm
  └   → /css21_dev/html4/border-left-width-applies-to-002.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-width-applies-to-001.htm
  └   → /css21_dev/html4/border-left-width-applies-to-001.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-color-applies-to-001.htm
  └   → /css21_dev/html4/border-top-color-applies-to-001.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-width-applies-to-003.htm
  └   → /css21_dev/html4/border-top-width-applies-to-003.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-width-applies-to-002.htm
  └   → /css21_dev/html4/border-top-width-applies-to-002.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-color-applies-to-003.htm
  └   → /css21_dev/html4/border-top-color-applies-to-003.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-applies-to-002.htm
  └   → /css21_dev/html4/border-top-applies-to-002.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-applies-to-001.htm
  └   → /css21_dev/html4/border-left-applies-to-001.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-left-applies-to-003.htm
  └   → /css21_dev/html4/border-left-applies-to-003.htm 013a961068b0ccf85b71b6a479cc98e9fc77e102
/css21_dev/html4/reference/border-left-applies-to-001-ref.htm cef4f31905a1f467d5997afb9f0898584bf7d55f
Testing 013a961068b0ccf85b71b6a479cc98e9fc77e102 == cef4f31905a1f467d5997afb9f0898584bf7d55f

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-color-applies-to-002.htm
  └   → /css21_dev/html4/border-top-color-applies-to-002.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-width-applies-to-001.htm
  └   → /css21_dev/html4/border-top-width-applies-to-001.htm 293911574a59c7f3947436264557174a97acc7a5
/css21_dev/html4/reference/ref-filled-black-96px-square.htm f73d1fa4091b541ed29c5bc4f99358740860d9e8
Testing 293911574a59c7f3947436264557174a97acc7a5 == f73d1fa4091b541ed29c5bc4f99358740860d9e8

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-applies-to-001.htm
  └   → /css21_dev/html4/border-top-applies-to-001.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4

  ▶ FAIL [expected PASS] /css21_dev/html4/border-top-applies-to-003.htm
  └   → /css21_dev/html4/border-top-applies-to-003.htm f89297eefbde67ac8aecc56f3c1babac2ff57c50
/css21_dev/html4/reference/border-bottom-applies-to-001-ref.htm a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
Testing f89297eefbde67ac8aecc56f3c1babac2ff57c50 == a8aa4b5a520585465f75aab08fdbe30bed6eb1f4
```

**Comment by glennw** [comment]
Interesting ... Gecko seems to have updated without any test failures. I'll investigate today.

**Comment by jdm** [comment]
My money is on https://github.com/servo/webrender/commit/635d1441d4cbf63dabaa7d1c50f2777ddcc80f85 since all of the failing tests have no borders drawn according to the reftest analyzer.

**Comment by jdm** [comment]
It may just be that we were drawing the desired borders in those tests for incorrect reasons before, and the new test results are a more accurate reflection of our table border inadequacies.

**Comment by glennw** [comment]
Disabled those tests due to https://github.com/servo/servo/issues/18999.

r? @jdm

**Pull request merged**